README.md

```diff
@@ -8,33 +8,103 @@ This crate provides a Rust implementation of the Schnorr signature scheme for th
 
 ## Library Structure
-The library is partitioned into two components:
+The library is partitioned into the following components:
 
-- **Keys**: Module containing the secret key structure for signing messages, and the public key & keypair structures used in verification.
-- **Signatures**: Module containing functions to verify the validity of Schnorr signatures.
-- **Gadgets**: Contains the Plonk gadgets for signature verification.
+- **Keys**: Module containing the secret key structure for signing messages, as well as the public key and the double public key structures used in verification.
+- **Signatures**: Module containing the standard and double signature structs as well as functions to verify the validity of Schnorr signatures and double Schnorr signatures.
+- **Gadgets**: Contains the Plonk gadgets for in-circuit verification of Schnorr signatures and double Schnorr Signatures.
 
-## Schnorr Sigma Protocol
+## Signature Scheme Description
 
-The signature scheme implemented within the Phoenix protocol is based on the Schnorr Sigma protocol, compiled alongside the Fiat–Shamir transformation, to serve as a non-interactive signature scheme. Specifically, the Phoenix protocol employs a variant that utilizes double keys, enabling the delegation of computational processes within the protocol's later stages.
+### Notation
 
-### Signature Scheme Description
+In the following:
+- Multiplication of a point $P$ by a scalar $s$ stands for adding $P$ $s$-times to itself.
+- $\mathbb{Z}_q$ the set of congruence classes modulo $q$
+- for prime $q$: $\mathbb{Z}_q^× =  \mathbb{Z}_q \setminus \overline{0}_q$, all congruence classes modulo $q$ without the congruence class of $0$ modulo $q$.
 
-The details of the signature scheme are as follows:
+### Single Signature
 
-- **Setup**: A secret key `sk` is sampled from the finite field `F_t`, and corresponding public keys are computed as `pk = sk * G` and `pk' = sk * G'`, where `G` and `G'` are generators of the JubJub elliptic curve group `J`. The public key pair is represented as `(pk, pk')`.
+#### Setup
 
-- **Signing Process**: To sign a message `m` using the secret key `sk`, a random scalar `r` is drawn from `F_t`. The commitment points `(R, R')` are calculated by multiplying `r` with the base points `G` and `G'`. A challenge `c` is derived by hashing the tuple `(m, R, R')`, and the response `u` is computed as `u = r - c * sk`. The signature is then the tuple `(R, R', u)`.
+In this library we implement our Schnorr signature scheme on the jubjub elliptic curve, specifically we have:
+- a finite field $\mathbb{Z}_q$ over prime $q$, in this implementation the field of scalar of the elliptic curve BLS12-381
+- an elliptic curve $E: \mathbb{Z}_q \rightarrow \mathbb{Z}_q$, in our case this is the jubjub elliptic curve
```

Comment:
- an elliptic curve $E / \mathbb{F}_q$, in our case this is the jubjub elliptic curve